kore/src/Kore/Step/Simplification/PredicateSubstitution.hs

```diff
@@ -120,26 +109,28 @@ simplify
                     assertDistinctVariables
                         (substitution <> simplifiedSubstitution)
                     (mergedPredicateSubstitution, _proof) <-
-                        mergePredicatesAndSubstitutions
+                        Monad.Trans.lift
```

Comment:
I know that you used returnPruned below, but using `liftPruned` here would make it easier to check that we always prune.